app/src/main/kotlin/com/wire/android/ui/userprofile/image/AvatarPicker.kt

```diff
@@ -9,56 +9,59 @@ import androidx.compose.foundation.layout.padding
 import androidx.compose.material3.ExperimentalMaterial3Api
-import androidx.compose.material3.Icon
-import androidx.compose.material3.IconButton
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.material3.TopAppBarDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import com.wire.android.R
 import com.wire.android.ui.common.ArrowRightIcon
-import com.wire.android.ui.common.BackNavigationIconButton
 import com.wire.android.ui.common.bottomsheet.MenuBottomSheetItem
 import com.wire.android.ui.common.bottomsheet.MenuItemIcon
 import com.wire.android.ui.common.bottomsheet.MenuModalSheetLayout
 import com.wire.android.ui.common.dimensions
-import com.wire.android.ui.common.imagepreview.BitmapState
 import com.wire.android.ui.common.imagepreview.BulletHoleImagePreview
+import com.wire.android.ui.common.imagepreview.PictureState
 import com.wire.android.ui.common.textfield.WirePrimaryButton
+import com.wire.android.ui.common.topappbar.BackNavigationIconButton
 import com.wire.android.ui.theme.wireTypography
-import com.wire.android.ui.userprofile.UserProfileViewModel
+import com.wire.android.util.getTempAvatarUri
 
 @OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class)
 @Composable
-//TODO: the input data for ProfileImageScreen will be decided later on after sync with Yamil
-fun AvatarPickerScreen(
-    viewModel: UserProfileViewModel
-) {
-    val state = rememberProfileImageState(viewModel.userProfileState.avatarBitmap)
+fun AvatarPickerScreen(viewModel: AvatarPickerViewModel) {
+    val context = LocalContext.current
+    val state = rememberAvatarPickerState()
+
+    // We need to launch an effect to update the initial avatar uri whenever the pickerVM updates successfully the raw image
+    LaunchedEffect(viewModel.avatarRaw) {
+        val currentAvatarUri = getTempAvatarUri(viewModel.avatarRaw ?: ByteArray(16), context)
+        state.avatarPickerFlow.pictureState = PictureState.Initial(currentAvatarUri)
+    }
 
-    AvatarPickerBottomSheet(
+    AvatarPickerContent(
         state = state,
         onCloseClick = {
-            when (val avatarImageState = state.picturePickerFlow.bitmapState) {
-                is BitmapState.BitmapPicked -> viewModel.changeUserAvatar(avatarImageState.bitmap, shouldNavigateBack = true)
-                is BitmapState.InitialBitmap -> viewModel.navigateBack()
+            if (state.avatarPickerFlow.pictureState is PictureState.Picked) {
+                viewModel.uploadNewPickedAvatarAndBack(state.avatarPickerFlow.pictureState.avatarUri, context)
+            } else {
+                viewModel.navigateBack()
             }
```

Comment:
I think we talked about this on the call, maybe I did not explain it correctly to you but here on the designs : 

![image](https://user-images.githubusercontent.com/15973138/157654606-7c14a47f-70da-485a-ac23-24d08f08cfbb.png)

the loading of the image is happening actually on the UserProfileScreen. I also discussed this approach with Vitor I think you took and the issues here are : 

if loading takes long time, it looks like screen freezes, we could add a spinner on AvatarPicker to indicate loading, but then lets say either user has no internet or it takes like 1 minutes, we have no option to go back to the previous screeen while loading as the back button is actually starting the upload.

Reply:
I know that the designs indicate that we should do the upload on the user profile screen, but actually from the technical point of view, it would make more sense to do the api call to upload the avatar on the picker screen. 

In any case, my idea was to just merge these changes now and fix the picker screen as it is not functional atm. Then once Wolfgang is back, we will need to have a talk with him, and decide on what's the best approach on where to upload it, and error handling, since the current design also doesn't allow the user to reset the picture to the original one once she/he has picked a new one. What do you think?

Reply:
sure, I am fine with it, just was thinking you did notice it